### ISSUE TITLE
Missing "name" and "version" keys in package.json and package-lock.json

### ISSUE THREAD

**Issue by x80486** [wireapp/wire-desktop#1692]
Is there any reason why `"name"` and `"version"` keys are missing in `package.json` – and eventually later on in `package-lock.json`?

According to [Specifics of npm's package.json handling](https://docs.npmjs.com/files/package.json):

> If you plan to publish your package, the _most_ important things in your `package.json` are the `name` and `version` fields as they will be required.

**Comment by ffflorian** [comment]
Yes:

> If you plan to publish your package, [...]

We are not publishing this package. So we don't need the fields. :wink:

**Comment by x80486** [comment]
Ha! I like it! But...since I'm trying to do a Flatpak for this one, can you put those 😇 — if it's not too much trouble.

**Comment by ConorIA** [comment]
@x80486, I got started with a Flatpak, but couldn't make it work. I _did_ workaround the version and name issue though using build commands. I'm attaching the files I had, perhaps they will be of some minor use to you. 

[wire flatpak.tar.gz](https://github.com/wireapp/wire-desktop/files/2334202/wire.flatpak.tar.gz)

**Comment by x80486** [comment]
I'm able to move (a little bit) forward with this one. I have the work [here](https://github.com/x80486/flathub/tree/com.wire.WireDesktop), in case you would like to give it a shot if you have some free time.

The Python script generates a `generated-sources.json` in order for Flatpak to build the artifact later on while being offline. I'm getting this error right now:

```sh
> @ preinstall /run/build/wire/main
> cd electron && npm install

npm info it worked if it ends with ok
npm info using npm@5.6.0
npm info using node@v8.11.4
npm info lifecycle wireinternal@~preinstall: wireinternal@
npm ERR! Error while executing:
npm ERR! /usr/bin/git ls-remote -h -t https://github.com/wireapp/node-addressbook.git
npm ERR! 
npm ERR! fatal: unable to access 'https://github.com/wireapp/node-addressbook.git/': Couldn't resolve host 'github.com'
npm ERR! 
npm ERR! exited with error code: 128
```

It's at first confusing because it looks like I don't have access to GitHub, but in reality is the Flatpak builder that never goes online – I know for sure because I ran into the same problem with another Electron application a few days ago.

Funny thing is, the `generated-sources-prod.json` that you gave me have that dependency:

```json
    "node-addressbook": {
      "version": "git+https://github.com/wireapp/node-addressbook.git#5799991759e017d92931075c5fd6eed9e66936b9",
      "optional": true,
      "requires": {
        "nan": "2.8.0"
      }
    },
```

...but the one I'm generating from the `.tar.gz` (release) file, does not. So you are applying _some magic_ that I don't know :open_mouth: ...I was expecting those files to match :no_good_man:

**Comment by ConorIA** [comment]
I haven't looked at this in some weeks. I do recall that the errors I was hitting were missing dependencies, even though I had thought that I had captured them all in the two package-lock.json files. I can't recall any magic in particular, but I think that there is something fishy going on in the `sed` command at the bottom of the generated-sources file. You seem to be much further ahead of the curve than I was as I fumbled though this. :worried:
